### ISSUE TITLE
Can not resolve IObjectResolver itself

### ISSUE THREAD

**Issue by neuecc** [hadashiA/VContainer#188]
when using `IObjectResolver` directly, can not resolver `IObjectResolver` itself.

```csharp
var builder = new ContainerBuilder(); 
var resolver = builder.Build();
var resolver2 = r.Resolve<IObjectResolver>(); // exception, not registered
```

I'm currently using the following code for unit testing.

```csharp
public class VContainerTest
{
    [Test]
    public void VContainerTestSimplePasses()
    {
        var builder = new ContainerBuilder();
        builder.Register<MyClass>(Lifetime.Transient);

        var resolver = builder.Build();

        // exception, can not resolve IObjectResolver...
        var mc = resolver.Resolve<MyClass>();
    }
}
public class MyClass
{
    readonly IObjectResolver resolver;
    public MyClass(IObjectResolver resolver)
    {
        this.resolver = resolver;
    }
}
```

Do you have any workaround?

**Comment by hadashiA** [comment]
In that case, nothing was registered by default..
Please try calling `builder.RegisterContainer()` ?  The `IObjectResolver` will be registered.

**Comment by neuecc** [comment]
Thanks, i've missed its api.
It completely works fine.

By the way, is there any reason why this is not the default?

**Comment by hadashiA** [comment]
> is there any reason why this is not the default?

During the initial development, I assumed Unity users around me who were not familiar with DI. That's why I had made these features as optional.

However, it no longer makes sense.
( If we use `LifetimeScope`,  `IObjectResolver` will be registered by default. )

Using ContainerBuilder directly was undocumented and I didn't care much about it,
but I think I'll change the default here too.

**Comment by hadashiA** [comment]
IObjectResolver registration is now the default.

Released in [v1.7.2](https://github.com/hadashiA/VContainer/releases/tag/1.7.2)
